test/samples/forall_simple.sol

```diff
@@ -0,0 +1,51 @@
+pragma solidity 0.8.4;
+
+/// #invariant forall (uint i in a) a[i] > 10;
+contract ForallSimple {
+	uint[] a;
+	function push(uint x) external {
+		a.push(x);
+	}
+
+	/// #if_succeeds forall (uint i in a) let z1 := x in let z2 := a[i] in z2 > z1;
+	function test1(uint[] memory a, uint x) public {
+	}
+
+	uint t = 0;
+	/// #if_succeeds forall (uint i in a) a[i] > old(t);
+	function test2(uint[] memory a, uint x) public {
+		t = x;
+	}
+
+	uint t1 = 0;
+	/// #if_succeeds old(forall (uint i in a) let x := a[i] in let y := t1 in x > y);
+	function test3(uint[] memory a, uint x) public {
+		t1 = x;
+	}
+
+	uint t2 = 0;
+	/// #if_succeeds forall (uint i in a) forall (uint j in a[i]) a[i][j] > old(t2);
+	function test4(uint[][] memory a, uint x) public {
+		t2 = x;
+	}
+
+	/// #if_succeeds forall (uint i in 1...5) i >= 1 && i < 5;
+	function test5() public {
+	}
+
+	/// #if_succeeds forall (uint i in 1...5) i > 1;
+	function test6() public {
+	}
+
+	/// #if_succeeds forall (uint i in 1...5) i < 4;
```

Comment:
@cd1m0 Is this supposed to hold?

Reply:
No - this is supposed to revert. There is a test in https://github.com/ConsenSys/scribble/pull/40/files#diff-615b432e8bcc686549a508d00ebfdf8bf8c2f846a9cbc0bb1a2a13c0c9e963d0 that checks that when calling `test7` after instrumentation it reverts.

Reply:
This test is just here to make sure the instrumentation  really is checking the ranges strictly.